CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -386,6 +386,10 @@ public static function sendMail($contactID, $values, $isTest = FALSE, $returnMes
         'receipt_date' => !$values['receipt_date'] ? NULL : date('YmdHis', strtotime($values['receipt_date'])),
       );
 
+      if (!empty($values['is_pay_later'])) {
```

Comment:
The issue here is that we have an unreliable value in 'is_pay_later' - if you check this PR https://github.com/civicrm/civicrm-core/pull/8159/files - it is addressing the unreliability - ie. the fact that is_pay_later is being overwritten based on the contribution page value.

In general the sendMail function suffers from leakage - so I prefer to be aggressive about setting values - ie without the !empty where possible. But, these are only quick notes - I'm going to look properly later